src/analysis/Environments/Environments.ts

```diff
@@ -135,15 +136,22 @@ export const Environments = (props: EnvironmentsProps): ReactNode => {
 
   const [shouldFilterByCreator, setShouldFilterByCreator] = useState(true);
 
+  // TODO: this function is a great candidate for breaking out into a separate hook.
```

Comment:
@msilva-broad I'm happy to separately try and work on refactoring some of this stuff to a dedicated hook. I thought it'd be good to first fix this bug in the short term, though.